### PR TITLE
dt-schema: 2022.1 -> 2022.12

### LINKED ISSUE #207837: dt-schema out of date
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: dt-schema

current version: 2022.1
desired version: 2022.12

###### Notify maintainers

maintainers: @sorki 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by Skallwar**
###### Description of changes
- Update dt-schema to 2022.12
- Add a patch for fixing the name of a new dependency `pylibfdt`. Nixos already have `libfdt` wich is the same just not the same name found on [Pypi](https://pypi.org/project/pylibfdt/)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Fixes #207837

### CODE REVIEW

**Review by sorki — APPROVED** [review]
Thank you!

**Comment by Skallwar** [comment]
I'm not familiar with Nixpkgs process, should I do something else to get this merge? I don't mind the wait, I just want to make sure that I did not forgot anything

File: pkgs/development/tools/dt-schema/default.nix [hunk 1]
```diff
@@ -7,23 +7,31 @@
...
-    sha256 = "sha256-G5KzuaMbbkuLK+cNvzBld1UwvExS6ZGVW2e+GXQRFMU=";
+    sha256 = "sha256-9TLUM6kV5QfUtCa371emcw0jwJOLaC+0hm+d+oCljsY=";
   };
 
+  patches = [
+    # Change name of pylibfdt to libfdt
+    ./fix_libfdt_name.patch
+  ];
+
   nativeBuildInputs = [ setuptools-scm git ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dt-schema/default.nix:27]:
```suggestion
  nativeBuildInputs = [ setuptools-scm ];
```
tested locally and not required

File: pkgs/development/tools/dt-schema/default.nix [hunk 2]
```diff
@@ -7,23 +7,31 @@
...
+
   propagatedBuildInputs = [
     setuptools
     ruamel-yaml
     jsonschema
     rfc3987
+    libfdt
   ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dt-schema/default.nix:36]:
```suggestion

  # has not tests in the sdist
  doCheck = false;

  pythonImportsCheck = [ "dtschema" ];

```

**Review by SuperSandro2000 — COMMENTED** [review]
Could you try switching to a source download, so that we can run tests?

File: pkgs/development/tools/dt-schema/default.nix [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
+  version = "2022.12";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-G5KzuaMbbkuLK+cNvzBld1UwvExS6ZGVW2e+GXQRFMU=";
+  src = fetchFromGitHub {
+    # inherit pname version;
+    owner = "devicetree-org";
+    repo = "dt-schema";
+    rev = "v" + version;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dt-schema/default.nix:21]:
```suggestion
    rev = "v${version}";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please rebase

File: pkgs/development/tools/dt-schema/default.nix [hunk 4]
```diff
@@ -1,31 +1,49 @@
...
 buildPythonPackage rec {
   pname = "dtschema";
-  version = "2022.1";
+  version = "2022.12";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-G5KzuaMbbkuLK+cNvzBld1UwvExS6ZGVW2e+GXQRFMU=";
+  src = fetchFromGitHub {
+    # inherit pname version;
```
[the remark this answers is not in the document]
Comment (Skallwar) [reply pkgs/development/tools/dt-schema/default.nix]: Fixed

File: pkgs/development/tools/dt-schema/default.nix (hunk 3 above)
Comment (Skallwar) [reply pkgs/development/tools/dt-schema/default.nix]: Fixed

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You need to add an entry into python-packages.nix.

**Comment by sorki** [comment]
I think that was done here https://github.com/NixOS/nixpkgs/pull/219511

**Comment by Skallwar** [comment]
> I think that was done here https://github.com/NixOS/nixpkgs/pull/219511

By looking at this I understand that my changes should be applied to `pkgs/development/python-modules/dtschema/default.nix` instead of `pkgs/development/tools/dt-schema/default.nix`. Am I understanding this correctly?

**Comment by SuperSandro2000** [comment]
dtschema got moved, so the new location is in python-modules.

**Comment by Skallwar** [comment]
Ok I will do this. Thanks for the help, I must have made a mistake while doing the rebase

File: pkgs/development/python-modules/dtschema/default.nix [hunk 5]
```diff
@@ -19,9 +20,14 @@ buildPythonPackage rec {
...
     repo = "dt-schema";
     rev = "refs/tags/v${version}";
-    hash = "sha256-wwlXIM/eO3dII/qQpkAGLT3/15rBLi7ZiNtqYFf7Li4=";
+    sha256 = "sha256-+wF6WdonZrkZEnlq/P6QeT3X7CMinxbapLa7q0t2zUc=";
   };
 
+  patches = [
+    # Change name of pylibfdt to libfdt
+    ./fix_libfdt_name.patch
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dtschema/default.nix:29]: Can you upstream this?

**Pull request merged**